Resources/Prototypes/Entities/Objects/Devices/encryption_keys.yml

```diff
@@ -0,0 +1,272 @@
+#EncryptionKeyComponent, EncryptionKeySystem
+- type: entity
+  abstract: true
+  parent: BaseItem
+  id: EncryptionKey
+  name: encryption key
+  description: A small cypher chip for headsets.
+  components:
+  - type: Item
+    sprite: Objects/Devices/encryption_keys.rsi
+  - type: EncryptionKey
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyCommon
+  name: passanger encryption key
+  description: An encryption key used by anyone.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Common
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyCargo
+  name: cargo encryption key
+  description: An encryption key used by supply employees.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Common
+    - Supply
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: cargo_cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyQM
+  name: qm encryption key
+  description: An encryption key used by boss of the cargo.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Command
+    - Common
+    - Supply
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: qm_cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyCentCom
+  name: central command encryption key
+  description: An encryption key used by captain's bosses.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Common
+    - Command
+    - CentCom
+    - Engineering
+    - Medical
+    - Science
+    - Security
+    - Service
+    - Supply
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: bin_cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyCommand
+  name: command encryption key
+  description: An encryption key used by crew's bosses.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Common
+    - Command
+    - Engineering
+    - Medical
+    - Science
+    - Security
+    - Service
+    - Supply
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: com_cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyEngineering
+  name: engineering encryption key
+  description: An encryption key used by engineers of this station.
+  components:
+  - type: EncryptionKey
+    channels:
+    - Common
+    - Engineering
+  - type: Sprite
+    sprite: Objects/Devices/encryption_keys.rsi
+    state: eng_cypherkey
+
+- type: entity
+  parent: EncryptionKey
+  id: EncryptionKeyCE
+  name: chief engineer's encryption key
+  description: An encryption key used by boss of this station.
```

Comment:
"boss of this station"? The chief engineer isn't the boss...

Reply:
He is ■-■